### ISSUE TITLE
deployment: curl localhost connection refused

### ISSUE THREAD

**Issue by jmugan** [tiangolo/full-stack-fastapi-postgresql#78]
Hi, I don't know a lot about networking. Can someone help? I did the deploy on an Amazon machine, and everything looked like it worked, all the services are running. But when I curl I get `connection refused`. I even get the connection refused from within the machine itself. Can someone point me in the right direction? Thanks so much. 

Here we see that netstat shows nothing is listening on port 80 after deployment 
![image](https://user-images.githubusercontent.com/7061275/69071979-8831fd80-09f0-11ea-98be-e6132d6a04f4.png)

**Comment by jmugan** [comment]
Okay, I had to do a bunch of stuff. Probably more than necessary since I don't know what I am doing. I did the following and it seems to work with `docker stack deploy`
- Upgrade to Traefik 2.0 and convert the labels to that format.
- specify ports for proxy in a file `docker-compose.deploy.ports.yml`
- specify a Traefik port for each service; e.g., `"traefik.http.services.flower.loadbalancer.server.port=5555"`
- add `"--providers.docker.swarmMode=true"` to `docker-compose.deploy.command.yml`
